src/plugins/EnergyRAPL.cpp

```diff
@@ -177,26 +189,40 @@ void EnergyRAPL::endStep(ParticleContainer* particleContainer, DomainDecompBase*
 void EnergyRAPL::readXML(XMLfileUnits& xmlconfig) {
 	xmlconfig.getNodeValue("writefrequency", _writeFrequency);
 	xmlconfig.getNodeValue("outputprefix", _outputprefix);
+	#ifdef ENABLE_MPI
+	xmlconfig.getNodeValue("per-host", _per_host);
+	#endif
 }
 
 void EnergyRAPL::outputEnergyJoule() {
 	double joule = _joule;
 #ifdef ENABLE_MPI
-	MPI_Reduce(&_joule, &joule, 1, MPI_DOUBLE, MPI_SUM, 0, MPI_COMM_WORLD);
-	if (_thisRank != 0) {
-		return;  // Only output on root rank
+	if (!_per_host) {
+		MPI_Reduce(&_joule, &joule, 1, MPI_DOUBLE, MPI_SUM, 0, MPI_COMM_WORLD);
+		if (_thisRank != 0) {
+			return;  // Only output on root rank
+		}
+	}
+	else if (!_thisRankShouldMeasure) {
+		return; // Only output on one rank per host
 	}
 #endif
 	std::string jouleStr;
 	std::stringstream sstream;
 	sstream << std::fixed << std::setprecision(6) << joule;
 	jouleStr = sstream.str();
 	if (_outputprefix.empty()) {
```

Comment:
You check here for `_outputprefix` but down in the code you rely on `getOutputFilename()` - knowing about it's internals. Not a dealbreaker here for me but generally I would think of having the actual output filename set in init as a class member instead. This would also save on some string operations in the output steps.